desktop/core/src/desktop/js/catalog/optimizer/ApiStrategy.ts

```diff
@@ -0,0 +1,188 @@
+const TOP_AGGS_URL = '/metadata/api/optimizer/top_aggs';
+const TOP_COLUMNS_URL = '/metadata/api/optimizer/top_columns';
+const TOP_FILTERS_URL = '/metadata/api/optimizer/top_filters';
+const TOP_JOINS_URL = '/metadata/api/optimizer/top_joins';
+const TOP_TABLES_URL = '/metadata/api/optimizer/top_tables';
+const TABLE_DETAILS_URL = '/metadata/api/optimizer/table_details';
+
+export default class ApiStrategy implements Optimizer {
+  analyzeCompatibility({
+    notebookJson,
+    snippetJson,
+    sourcePlatform,
+    targetPlatform,
+    silenceErrors
+  }: CompatibilityOptions): CancellablePromise<unknown> {
+    return post<unknown>(
+      COMPATIBILITY_URL,
+      {
+        notebook: notebookJson,
+        snippet: snippetJson,
+        sourcePlatform,
+        targetPlatform
+      },
+      { silenceErrors }
+    );
+  }
+
+  analyzeRisk({
+    notebookJson,
+    snippetJson,
+    silenceErrors
+  }: RiskOptions): CancellablePromise<OptimizerRisk> {
+    return post<OptimizerRisk>(
+      RISK_URL,
+      {
+        notebook: notebookJson,
+        snippet: snippetJson
+      },
+      { silenceErrors }
+    );
+  }
+
+  analyzeSimilarity({
+    notebookJson,
+    snippetJson,
+    sourcePlatform,
+    silenceErrors
+  }: SimilarityOptions): CancellablePromise<unknown> {
+    return post<unknown>(
+      SIMILARITY_URL,
+      {
+        notebook: notebookJson,
+        snippet: snippetJson,
+        sourcePlatform
+      },
+      { silenceErrors }
+    );
+  }
+
+  fetchPopularity({
+    paths,
+    silenceErrors
+  }: PopularityOptions): CancellablePromise<OptimizerResponse> {
+    let url, data;
+
+    if (paths.length === 1 && paths[0].length === 1) {
+      url = TOP_TABLES_URL;
+      data = {
+        database: paths[0][0]
+      };
+    } else {
+      url = TOP_COLUMNS_URL;
+      data = {
+        dbTables: JSON.stringify(paths.map(path => path.join('.')))
+      };
+    }
+
+    return post<OptimizerResponse>(url, data, {
+      silenceErrors,
+      handleSuccess: (response, resolve, reject) => {
+        if (successResponseIsError(response)) {
+          reject(extractErrorMessage(response));
+        } else {
+          response.hueTimestamp = Date.now();
+          resolve(response);
+        }
+      }
+    });
+  }
+
+  fetchTopAggs(options: PopularityOptions): CancellablePromise<TopAggs> {
+    return genericOptimizerMultiTableFetch<TopAggs>(options, TOP_AGGS_URL);
+  }
+
+  fetchTopColumns(options: PopularityOptions): CancellablePromise<TopColumns> {
+    return genericOptimizerMultiTableFetch<TopColumns>(options, TOP_COLUMNS_URL);
+  }
+
+  fetchTopFilters(options: PopularityOptions): CancellablePromise<TopFilters> {
+    return genericOptimizerMultiTableFetch<TopFilters>(options, TOP_FILTERS_URL);
+  }
+
+  fetchTopJoins(options: PopularityOptions): CancellablePromise<TopJoins> {
+    return genericOptimizerMultiTableFetch<TopJoins>(options, TOP_JOINS_URL);
+  }
+
```

Comment:
Note for later: we could have the token or snippet API at some point here (e.g. full CREATE TABLE ...., multi tokens... suggestions)